src/PhpPact/Consumer/Driver/InteractionPart/AbstractInteractionPartDriver.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace PhpPact\Consumer\Driver\InteractionPart;
+
+use PhpPact\Consumer\Driver\Body\InteractionBodyDriver;
+use PhpPact\Consumer\Driver\Body\InteractionBodyDriverInterface;
+use PhpPact\Consumer\Driver\Enum\InteractionPart;
+use PhpPact\Consumer\Model\Interaction;
+use PhpPact\FFI\ClientInterface;
+
+abstract class AbstractInteractionPartDriver
+{
+    private InteractionBodyDriverInterface $bodyDriver;
+
+    public function __construct(
+        protected ClientInterface $client,
+        ?InteractionBodyDriverInterface $bodyDriver = null
+    ) {
+        $this->bodyDriver = $bodyDriver ?? new InteractionBodyDriver($client);
+    }
+
+    protected function withBody(Interaction $interaction, InteractionPart $part): self
+    {
+        $this->bodyDriver->registerBody($interaction, $part);
+
+        return $this;
```

Comment:
This feels like it enforces, or expects a sequence of calls to initialize an object to a state ready for use.
`withBody` and `withHeaders` could be available, but should be initializable without these calls, I think. I won't block on it, but it could manifest in folks becoming confused.

Reply:
I will check and remove `$this` if these methods can be call in any order in a later PR

Reply:
PR https://github.com/pact-foundation/pact-php/pull/492